crates/matrix-sdk-crypto/src/identities/device.rs

```diff
@@ -780,13 +780,10 @@ impl ReadOnlyDevice {
 
         if let Some(mut session) = session {
             let message = session.encrypt(self, event_type, content, message_id).await?;
-
             trace!("Successfully encrypted an event");
-
             Ok((session, message))
         } else {
-            warn!("Trying to encrypt an event for a device, but no Olm session is found.",);
```

Comment:
this happens quite a lot, but we'll normally have logged about it during `get_missing_sessions` or so. We really don't want to be logging a warning for every single device that hits this path.